libs/npm/core/src/components/grid-table/Pager.tsx

```diff
@@ -8,95 +8,112 @@ import { Text } from '../form/text';
   nextPage: () => void;
+  /** Event when going to previous page. */
   previousPage: () => void;
+  /** Event when page size changes. */
   setPageSize: (pageSize: number) => void;
 }
 
+/**
+ * Provides a manually change the page.
+ * @param param0 Component properties.
+ * @returns Component.
+ */
 export const Pager: React.FC<IPagerProps> = ({
   pageIndex,
   pageSize,
   pageCount,
-  pageLimit = 5,
+  pageLimit = 10,
   canPreviousPage,
   showPageGroup = false,
   canNextPage,
   pageOptions = [],
+  pageSizeOptions = {},
   gotoPage,
   nextPage,
   previousPage,
   setPageSize,
-  manualPageSize,
 }) => {
   const { search } = useLocation();
   let query = React.useMemo(() => new URLSearchParams(search), [search]);
   let [searchParams, setSearchParams] = useSearchParams(query);
 
-  React.useEffect(() => {
-    if (manualPageSize) {
-      const qty = searchParams.get(QTY);
-      const cachedQty = localStorage.getItem(QTY);
+  const [quantity, setQuantity] = React.useState(pageSize);
 
-      if (!!qty && qty !== pageSize.toString()) {
-        setPageSize(parseInt(qty));
-      }
+  const {
+    show: showPageSize = true,
+    fromUrl = true,
+    fromLocalStorage = false,
+    key = QTY,
+  } = pageSizeOptions;
 
-      if (!qty && !!cachedQty && cachedQty !== pageSize.toString()) {
-        setPageSize(parseInt(cachedQty));
-        setSearchParams({ ...searchParams, qty: cachedQty });
+  const qtyUrl = Number(searchParams.get(key));
+  const qtyLS = Number(localStorage.getItem(key));
+
+  React.useEffect(() => {
+    if (fromUrl && fromLocalStorage) {
+      if (!isNaN(qtyUrl) && !!qtyUrl && qtyUrl !== qtyLS) {
+        if (fromLocalStorage && qtyUrl !== qtyLS) localStorage.setItem(key, qtyUrl.toString());
+      } else if (!isNaN(qtyLS) && !!qtyLS) {
+        setSearchParams({ ...searchParams, [key]: qtyLS.toString() });
       }
     }
-    // Extract quantity from URL or cookie the first time loading.
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+    // Extract quantity from URL and update local storage, or extract from local storage and update the URL the first time loading.
+  }, [fromLocalStorage, fromUrl, key, qtyLS, qtyUrl, searchParams, setSearchParams]);
 
   React.useEffect(() => {
-    if (manualPageSize) {
-      const qty = searchParams.get(QTY);
-      if (!!qty) {
-        if (pageSize !== parseInt(qty)) setPageSize(parseInt(qty));
-      }
+    // The URL has changed, update the page number.
+    if (fromUrl) {
+      if (!isNaN(qtyUrl) && !!qtyUrl && qtyUrl !== pageSize) setPageSize(qtyUrl);
+    } else if (fromLocalStorage) {
+      if (!isNaN(qtyLS) && !!qtyLS && qtyLS !== pageSize) setPageSize(qtyLS);
     }
-  }, [manualPageSize, pageSize, searchParams, setPageSize]);
+    // Change the page size if the URL or local storage have a different value.
+  }, [fromUrl, fromLocalStorage, pageSize, setPageSize, qtyUrl, qtyLS]);
+
+  React.useEffect(() => {
```

Comment:
Need to manage our own state so that the user can make changes without causing tons of refreshes.